app/src/main/java/com/codecademy/movieapp/HomeListAdapter.kt

```diff
@@ -0,0 +1,43 @@
+package com.codecademy.movieapp
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import android.widget.ImageView
+import android.widget.TextView
+
+class HomeListAdapter constructor(context: Context, private val resource: Int, private val itemList: Array<String>?)
+    : ArrayAdapter<HomeListAdapter.ItemHolder>(context, resource){
+
+    override fun getCount(): Int {
+        return if (this.itemList != null) this.itemList.size else 0
+    }
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+        var convertView = convertView
+
+        val holder: ItemHolder
+        if (convertView == null) {
+            convertView = LayoutInflater.from(context).inflate(resource, null)
+            holder = ItemHolder()
+            holder.name = convertView!!.findViewById(R.id.home_textView)
+            holder.icon = convertView.findViewById(R.id.image_btn)
+            convertView.tag = holder
+        } else {
+            holder = convertView.tag as ItemHolder
+        }
+
+        holder.name!!.text = this.itemList!![position]
+        holder.icon!!.setImageResource(R.mipmap.ic_launcher)
+
+        return convertView
+    }
+
+    class ItemHolder {
+        var name: TextView? = null
+        var icon: ImageView? = null
+    }
+
+}
```

Comment:
ArrayAdapter is for a ListView. 
For a RecyclerView, we should do like this:
```suggestion
class HomeListAdapter constructor(private val itemList: Array<String>) :
    RecyclerView.Adapter<HomeListAdapter.ViewHolder>() {

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val nameTextView: TextView = view.findViewById(R.id.home_textView)
        val iconImageButton: ImageView = view.findViewById(R.id.image_btn)
    }

    override fun getItemCount() = itemList.size

    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
        val view =
            LayoutInflater.from(viewGroup.context).inflate(R.layout.list_item, viewGroup, false)

        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.nameTextView.text = itemList[position]
        holder.iconImageButton.setImageResource(R.mipmap.ic_launcher)
    }
}
```

Please take a look at the Android doc https://developer.android.com/guide/topics/ui/layout/recyclerview